osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -146,54 +156,143 @@ public Storage LocateStableStorage()
                 Logger.Error(e, "Stable installation could not be found; disabling file based IPC");
             }
 
-            return Storage;
+            return IPCStorage;
         }
 
+        private static bool checkExists(string p) => File.Exists(Path.Combine(p, "ipc.txt"));
+
         private string findStablePath()
         {
-            static bool checkExists(string p) => File.Exists(Path.Combine(p, "ipc.txt"));
-
             string stableInstallPath = string.Empty;
 
             try
             {
-                try
+                List<Func<string>> stableFindMethods = new List<Func<string>>
                 {
-                    stableInstallPath = Environment.GetEnvironmentVariable("OSU_STABLE_PATH");
+                    findFromJsonConfig,
+                    findFromEnvVar,
+                    findFromRegistry,
+                    findFromLocalAppData,
+                    findFromDotFolder
+                };
+
+                foreach (var r in stableFindMethods)
+                {
+                    stableInstallPath = r.Invoke();
 
-                    if (checkExists(stableInstallPath))
+                    if (stableInstallPath != null)
+                    {
                         return stableInstallPath;
-                }
-                catch
-                {
+                    }
                 }
 
-                try
-                {
-                    using (RegistryKey key = Registry.ClassesRoot.OpenSubKey("osu"))
-                        stableInstallPath = key?.OpenSubKey(@"shell\open\command")?.GetValue(string.Empty).ToString().Split('"')[1].Replace("osu!.exe", "");
+                return stableInstallPath;
```

Comment:
Nit, but I'd probably explicitly return null/empty string here for clarity (right now it looks a little weird, as if it somehow found a path after trying everything and failing).